exp/controllers/azureasomanagedcluster_controller.go

```diff
@@ -83,6 +86,25 @@ func (r *AzureASOManagedClusterReconciler) SetupWithManager(ctx context.Context,
 				infracontroller.ClusterUpdatePauseChange(log),
 			),
 		).
+		Watches(
+			&infrav1exp.AzureASOManagedControlPlane{},
+			handler.EnqueueRequestsFromMapFunc(asoManagedControlPlaneToManagedClusterMap(r.Client)),
+			builder.WithPredicates(
+				predicates.ResourceHasFilterLabel(log, r.WatchFilterValue),
+				predicate.Funcs{
+					CreateFunc: func(ev event.CreateEvent) bool {
+						controlPlane := ev.Object.(*infrav1exp.AzureASOManagedControlPlane)
+						return !controlPlane.Status.ControlPlaneEndpoint.IsZero()
+					},
+					UpdateFunc: func(ev event.UpdateEvent) bool {
+						oldControlPlane := ev.ObjectOld.(*infrav1exp.AzureASOManagedControlPlane)
+						newControlPlane := ev.ObjectNew.(*infrav1exp.AzureASOManagedControlPlane)
+						return oldControlPlane.Status.ControlPlaneEndpoint !=
+							newControlPlane.Status.ControlPlaneEndpoint
+					},
```

Comment:
For my own understanding, why is `CreateFunc` and `UpdateFunc` needed for this Watch? I don't see them used anywhere else in CAPZ.

Reply:
This managed cluster controller only cares about AzureASOManagedControlPlane resources whose control plane endpoints have changed, and these functions encode that logic. They're invoked by controller-runtime, not by CAPZ directly.